src/AggregateFunctions/AggregateFunctionCountEqualRanges.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <AggregateFunctions/IAggregateFunction.h>
+#include <Columns/ColumnsNumber.h>
+#include <Common/SipHash.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+
+
+namespace DB
+{
+
+struct AggregateFunctionCountEqualRangesData
+{
+    UInt64 count = 0;
+    Field value;
+};
+
+class AggregateFunctionCountEqualRanges final : public IAggregateFunctionDataHelper<
+    AggregateFunctionCountEqualRangesData,
+    AggregateFunctionCountEqualRanges>
+{
+public:
+    AggregateFunctionCountEqualRanges(const DataTypes & arguments_, const Array & params_)
+        : IAggregateFunctionDataHelper<
+            AggregateFunctionCountEqualRangesData,
+            AggregateFunctionCountEqualRanges> {arguments_, params_}
+    {
+        // nothing
+    }
+
+    String getName() const override
+    {
+        return "countEqualRanges";
+    }
+
+    DataTypePtr getReturnType() const override
+    {
+        return std::make_shared<DataTypeUInt64>();
+    }
+
+    void ALWAYS_INLINE add(AggregateDataPtr place, const IColumn ** columns, size_t row_num, Arena *) const override
+    {
+        const Field & value = (*columns[0])[row_num];
+
+        if (!this->data(place).count || this->data(place).value != value)
```

Comment:
The definition of segment is confusing here. It relies on the order of value delivery.